### ISSUE TITLE
`each` helper does not work with objects with keys that contain slashes

### ISSUE THREAD

**Issue by softprops** [sunng87/handlebars-rust#152]
I was trying use this library for a hackathon to build some tooling around code generation with open api documents. open api documents have an object "paths" that has keys that present paths "/foo" that map to values which represent information about that paths api.

I ran into a problem with `each` helper where the binding of "this" was incorrect, defaulting to the top level context

I tested a small example to see keys that contained slashes where even supported [here](http://tryhandlebarsjs.com/) and it looks like they should be

template

```handlebars
{{#each .}}
  {{@key}} {{this}}
{{/each}}
```

context 

```json
{ 
  "/foo": "bar",
"/baz": "boom",
}
```

expected output

```
/foo bar
/baz boom
```

here's a minimal example to demonstrate

```rust
        let mut ctx = ::std::collections::BTreeMap::new();
        ctx.insert("/foo", "bar");
        ctx.insert("/baz", "boom");
        let mut hbs = ::handlebars::Handlebars::new();
        hbs.register_template_string("t0", "{{#each .}}
    {{@key}} {{this}}
  {{/each}}").unwrap();
        println!("{}", hbs.render("t0", &ctx).unwrap());
```

this prints

```
    /baz [object]

    /foo [object]
```

I have a strong hunch the source of the problem is either the path set [here](https://github.com/sunng87/handlebars-rust/blob/master/src/helpers/helper_each.rs#L80) or where ever that path gets interpreted. I'm happy to jump in at help if you can point me in the write direction.

**Comment by softprops** [comment]
glancing through the code, [this](https://github.com/sunng87/handlebars-rust/blob/76a06dde969bb9a3d5cbef8c60bae4d0dd5a7d71/src/render.rs#L296) may also be part of the problem.

if the path it set to the string "foo/bar.[/baz]" and you're later splitting that string on either `.` or `/`, I could see how that would be problematic!

**Comment by sunng87** [comment]
Good catch! I'll look into this issue later.

**Comment by softprops** [comment]
Thanks so much. I have one more day for the hackathon so I can maybe teach myself to fish.

**Comment by sunng87** [comment]
I can only look into this issue during weekends. So if you are in hurry, feel free to correct it in your way. Patches are always welcomed.

**Comment by sunng87** [comment]
This has been fixed in 0.26.2 release.

**Comment by softprops** [comment]
Oh snap! Thanks for the fast turn around @sunng87. I learned a bit about how this lib works by looking at that fix. Good stuff!
